content/artifacts/highlight-active-artifact-planets/plugin.js

```diff
@@ -0,0 +1,169 @@
+// Display the planet with active artifact, specially for cannon.
+// When cannon first activated, only a thin circle can be seen around the planet.
+// As time goes by, a thick arc will appears, and it will extend longer and longer.
+// The thick arc become a circle when the cannon ready.
+
+import { 
+  BiomeNames
+} from "https://cdn.skypack.dev/@darkforest_eth/types";
+
+import {
+  coords
+} from 'https://plugins.zkga.me/utils/utils.js';
+
+import {
+  ArtifactType,
+  ArtifactTypeNames
+} from "https://cdn.skypack.dev/@darkforest_eth/types";
+
+const EXCEPT_ARTIFACT_TYPES = [ArtifactType.BlackDomain, ArtifactType.BloomFilter, ArtifactType.Unknown];
+const ARTIFACT_TYPES = Object.values(ArtifactType).filter((type) => EXCEPT_ARTIFACT_TYPES.indexOf(type) === -1).map((type) => ({ value: type, text: ArtifactTypeNames[type] }));
+
+
+function getPlayerColor(ethAddress) {
+  return df.getProcgenUtils().getPlayerColor(ethAddress);
+}
+
+
+function drawArc(ctx, planet, artifact, color){
+
+  let percent = 1;
+  if (artifact.artifactType === ArtifactType.PhotoidCannon) {
+    let timePass = Math.floor(Date.now() / 1000 - artifact.lastActivated);
+    percent = timePass / ui.gameManager.contractConstants.PHOTOID_ACTIVATION_DELAY;
+  }
+
+  const viewport = ui.getViewport();
+  ctx.strokeStyle = color;
+  ctx.lineWidth = 5;
+  if (!planet.location) {
+    return false;
+  }
+  const {x, y} = viewport.worldToCanvasCoords(planet.location.coords);
+  const range = ui.getRadiusOfPlanetLevel(planet.planetLevel) * 4;
+  const trueRange = viewport.worldToCanvasDist(range);
+  ctx.beginPath();
+  ctx.arc(x, y, trueRange, 0, 2 * Math.PI * percent);
+  ctx.stroke();
+  if (artifact.artifactType === ArtifactType.PhotoidCannon && percent < 0.99) {
+    ctx.lineWidth = 1;
+    ctx.beginPath();
+    ctx.arc(x, y, trueRange, 0, 2 * Math.PI);
+    ctx.stroke();
+  }
+  return true;
+}
+
+
+class Plugin {
+  constructor() {
+    this.message = document.createElement('div');
+    this.showGroup = 'My';
+    this.showArifactTypes = [ArtifactType.PhotoidCannon];
```

Comment:
Please can `this.showArifactTypes` be changed to `this.showArtifactTypes` throughout this file?

Other than that, looks good!

Reply:
Thanks for the review, you are right